### ISSUE TITLE
The periodic build may be executed two or three times unintentionally

### ISSUE THREAD

**Issue by sugarnaoming** [screwdriver-cd/screwdriver#2311]
**What happened**:
The periodic build may be executed two or three times unintentionally.

**What you expected to happen**:
I want the periodic build to be executed exactly once.

**How to reproduce it**:
___I have not yet been able to establish a way to reproduce it.___
This happens in a periodic build where the  `workflowGraph` is included in the `periodicBuildConfig`(in redis database). However, it is not guaranteed to occur even if the `workflowGraph` is included.
At the same time, the number of 504 errors from queue-service increased.

**Comment by kkisic** [comment]
After investigating in detail, we found that it tends to occur when a large number of archived jobs are included in the same pipeline.
We also observed the following phenomena in such a pipeline.
- After committing to GitHub, there is a delay of several minutes before the build is created.
- `Request failed: POST https://<API_URL>/v4/events` occurs when creating the build from the UI.

I think the reason is that [this fix](https://github.com/screwdriver-cd/models/compare/v27.51.1...v27.51.2) handles it as follows.
When `POST /v4/events` is called, pipeline sync is called and it tries to call `DELETE /v1/queue/message?type=periodic` for all archived jobs in that pipeline.
As a result, the load on API seems to increase and it takes longer to create the build.

I think [this](https://github.com/screwdriver-cd/models/blob/2554d4adaa9a41b20ba5edcc213c28c517808f43/lib/job.js#L257) can be solved by modifying this part as follows.
```
if ((!newPeriodic && oldPeriodic) || (!newIsJobEnabled && oldIsJobEnabled) || (newIsJobArchived && !oldIsJobArchived))
```
